src/app/courses/courses-search/courses-search.component.html

```diff
@@ -2,7 +2,7 @@
   <mat-form-field appearance="outline" class="search__input">
     <mat-icon matPrefix class="icon big search__icon">search</mat-icon>
     <mat-label>Text to search</mat-label>
-    <input matInput [(ngModel)]="searchValue">
+    <input matInput [(ngModel)]="searchValue" (keypress)="handleKeyPress($event)">
```

Comment:
You don't need ngModel in your particular case here.
You are able to get the value from the event.